notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/NotificationEventConfig.java

```diff
@@ -0,0 +1,205 @@
+/**
+ * Copyright (C) 2018-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.streamplatform.streamregistry.core.events;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+import com.google.common.base.Preconditions;
+
+import io.confluent.kafka.serializers.KafkaAvroSerializer;
+import io.confluent.kafka.serializers.KafkaAvroSerializerConfig;
+
+import org.apache.avro.specific.SpecificRecord;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.kafka.core.DefaultKafkaProducerFactory;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.core.ProducerFactory;
+
+import com.expediagroup.streamplatform.streamregistry.core.events.handlers.SchemaEventHandlerForKafka;
+import com.expediagroup.streamplatform.streamregistry.model.Schema;
+
+@Slf4j
+@Configuration
+@EnableConfigurationProperties({NotificationEventConfig.NewTopicProperties.class, NotificationEventConfig.SchemaParserProperties.class})
+public class NotificationEventConfig {
+    public static final String KAFKA_SCHEMA_REGISTRY_URL_PROPERTY = "notification.events.kafka.schema.registry.url";
```

Comment:
Should we contain all configuration options under `streamRegistry` root?

Reply:
Do you mean something like `stream-registry.notification.events.kafka.schema.registry.url`?

Reply:
Or to define all possible configs in let say `com.expediagroup.streamplatform.streamregistry.StreamRegistryApp`, or `com.expediagroup.streamplatform.streamregistry.StreamRegistryAutoConfiguration`?
If that's accurate; lets define a place to set them